File: OPENAPI-REST-API/swagger-client/dart/lib/model/server_order_post_request.dart
part of swagger.api;

class ServerOrderPostRequest {
  /* CPU configuration id from config_li.cpu_li. Constrains valid hd/memory options. */
  int cpu = null;
/* Hard-drive configuration id(s) from config_li.hd_li[cpu]. Accepts a single id or an array of ids. */
  OneOfServerOrderPostRequestHd hd = null;
/* Memory configuration id from config_li.memory_li[cpu]. */
  int memory = null;
/* Bandwidth configuration id from config_li.bandwidth_li. */
  int bandwidth = null;
/* IP block configuration id from config_li.ips_li. */
  int ips = null;
/* Operating System configuration id from config_li.os_li. */
  int os = null;
/* Control Panel configuration id from config_li.cp_li. */
  int cp = null;
/* RAID configuration id from config_li.raid_li. */
  int raid = null;
/* Region id from the regions list. */
  int region = null;
/* Server hostname. Must pass valid_hostname(). */
  String servername = null;
/* Root password for the server. */
  String rootpass = null;
/* Terms-of-service acceptance. Must be true to place the order. */
  bool tos = null;
/* Optional free-form order comment. */
  String comment = "";

  ServerOrderPostRequest();

  @override
  String toString() {
    return 'ServerOrderPostRequest[cpu=$cpu, hd=$hd, memory=$memory, bandwidth=$bandwidth, ips=$ips, os=$os, cp=$cp, raid=$raid, region=$region, servername=$servername, rootpass=$rootpass, tos=$tos, comment=$comment, ]';
  }

  ServerOrderPostRequest.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    cpu = json['cpu'];
    hd = new OneOfServerOrderPostRequestHd.fromJson(json['hd']);
    memory = json['memory'];
    bandwidth = json['bandwidth'];
    ips = json['ips'];
    os = json['os'];
    cp = json['cp'];
    raid = json['raid'];
    region = json['region'];
    servername = json['servername'];
    rootpass = json['rootpass'];
    tos = json['tos'];
    comment = json['comment'];
  }

  Map<String, dynamic> toJson() {
    return {
      'cpu': cpu,
      'hd': hd,
      'memory': memory,
      'bandwidth': bandwidth,
      'ips': ips,
      'os': os,
      'cp': cp,
      'raid': raid,
      'region': region,
      'servername': servername,
      'rootpass': rootpass,
      'tos': tos,
      'comment': comment
     };
  }

  static List<ServerOrderPostRequest> listFromJson(List<dynamic> json) {
    return json == null ? new List<ServerOrderPostRequest>() : json.map((value) => new ServerOrderPostRequest.fromJson(value)).toList();
  }

  static Map<String, ServerOrderPostRequest> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, ServerOrderPostRequest>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new ServerOrderPostRequest.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/myadmin-client-csharp-dotnet2/Model/ServerOrderPostRequest.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// Request body to place a custom dedicated server order. All ids come from getNewServer&#x27;s config_li / regions.
  /// </summary>
  [DataContract]
  public class ServerOrderPostRequest {
    /// <summary>
    /// CPU configuration id from config_li.cpu_li. Constrains valid hd/memory options.
    /// </summary>
    /// <value>CPU configuration id from config_li.cpu_li. Constrains valid hd/memory options.</value>
    [DataMember(Name="cpu", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "cpu")]
    public int? Cpu { get; set; }

    /// <summary>
    /// Hard-drive configuration id(s) from config_li.hd_li[cpu]. Accepts a single id or an array of ids.
    /// </summary>
    /// <value>Hard-drive configuration id(s) from config_li.hd_li[cpu]. Accepts a single id or an array of ids.</value>
    [DataMember(Name="hd", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "hd")]
    public OneOfServerOrderPostRequestHd Hd { get; set; }

    /// <summary>
    /// Memory configuration id from config_li.memory_li[cpu].
    /// </summary>
    /// <value>Memory configuration id from config_li.memory_li[cpu].</value>
    [DataMember(Name="memory", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "memory")]
    public int? Memory { get; set; }

    /// <summary>
    /// Bandwidth configuration id from config_li.bandwidth_li.
    /// </summary>
    /// <value>Bandwidth configuration id from config_li.bandwidth_li.</value>
    [DataMember(Name="bandwidth", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "bandwidth")]
    public int? Bandwidth { get; set; }

    /// <summary>
    /// IP block configuration id from config_li.ips_li.
    /// </summary>
    /// <value>IP block configuration id from config_li.ips_li.</value>
    [DataMember(Name="ips", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ips")]
    public int? Ips { get; set; }

    /// <summary>
    /// Operating System configuration id from config_li.os_li.
    /// </summary>
    /// <value>Operating System configuration id from config_li.os_li.</value>
    [DataMember(Name="os", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "os")]
    public int? Os { get; set; }

    /// <summary>
    /// Control Panel configuration id from config_li.cp_li.
    /// </summary>
    /// <value>Control Panel configuration id from config_li.cp_li.</value>
    [DataMember(Name="cp", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "cp")]
    public int? Cp { get; set; }

    /// <summary>
    /// RAID configuration id from config_li.raid_li.
    /// </summary>
    /// <value>RAID configuration id from config_li.raid_li.</value>
    [DataMember(Name="raid", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "raid")]
    public int? Raid { get; set; }

    /// <summary>
    /// Region id from the regions list.
    /// </summary>
    /// <value>Region id from the regions list.</value>
    [DataMember(Name="region", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "region")]
    public int? Region { get; set; }

    /// <summary>
    /// Server hostname. Must pass valid_hostname().
    /// </summary>
    /// <value>Server hostname. Must pass valid_hostname().</value>
    [DataMember(Name="servername", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "servername")]
    public string Servername { get; set; }

    /// <summary>
    /// Root password for the server.
    /// </summary>
    /// <value>Root password for the server.</value>
    [DataMember(Name="rootpass", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "rootpass")]
    public string Rootpass { get; set; }

    /// <summary>
    /// Terms-of-service acceptance. Must be true to place the order.
    /// </summary>
    /// <value>Terms-of-service acceptance. Must be true to place the order.</value>
    [DataMember(Name="tos", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tos")]
    public bool? Tos { get; set; }

    /// <summary>
    /// Optional free-form order comment.
    /// </summary>
    /// <value>Optional free-form order comment.</value>
    [DataMember(Name="comment", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "comment")]
    public string Comment { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class ServerOrderPostRequest {\n");
      sb.Append("  Cpu: ").Append(Cpu).Append("\n");
      sb.Append("  Hd: ").Append(Hd).Append("\n");
      sb.Append("  Memory: ").Append(Memory).Append("\n");
      sb.Append("  Bandwidth: ").Append(Bandwidth).Append("\n");
      sb.Append("  Ips: ").Append(Ips).Append("\n");
      sb.Append("  Os: ").Append(Os).Append("\n");
      sb.Append("  Cp: ").Append(Cp).Append("\n");
      sb.Append("  Raid: ").Append(Raid).Append("\n");
      sb.Append("  Region: ").Append(Region).Append("\n");
      sb.Append("  Servername: ").Append(Servername).Append("\n");
      sb.Append("  Rootpass: ").Append(Rootpass).Append("\n");
      sb.Append("  Tos: ").Append(Tos).Append("\n");
      sb.Append("  Comment: ").Append(Comment).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
